### PR TITLE
sql: deflake TestCancelQueryPermissions

### LINKED ISSUE #63547: sql: TestCancelQueryPermissions flaking
The `TestCancelQueryPermissions` flaked with:

```
=== RUN   TestCancelQueryPermissions/unpermissioned_users_cannot_cancel_other_users
panic: Fail in goroutine after TestCancelQueryPermissions/non-admins_with_CANCELQUERY_can_cancel_non-admins has completed

goroutine 365437 [running]:
testing.(*common).Fail(0xc00757e780)
	/usr/local/go/src/testing/testing.go:688 +0x125
testing.(*common).FailNow(0xc00757e780)
	/usr/local/go/src/testing/testing.go:710 +0x2b
testing.(*common).Fatalf(0xc00757e780, 0x47c9e4c, 0x1c, 0xc00ba7a4a0, 0x2, 0x2)
	/usr/local/go/src/testing/testing.go:806 +0x93
github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).ExpectErr(0xc002060fa8, 0x560fca0, 0xc00757e780, 0x47b8f45, 0x18, 0x47b6bed, 0x18, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:97 +0x1c5
github.com/cockroachdb/cockroach/pkg/sql_test.TestCancelQueryPermissions.func2.1(0xc0136594f0, 0xc002586340, 0xc00757e780)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/run_control_test.go:327 +0xae
created by github.com/cockroachdb/cockroach/pkg/sql_test.TestCancelQueryPermissions.func2
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/run_control_test.go:318 +0x13d
```

Flake can be found at - https://teamcity.cockroachdb.com/viewLog.html?buildId=2876992&buildTypeId=Cockroach_UnitTests

Not skipping for now, as we have only seen one instance of it.

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/cockroach/issues/63547

Release note: None

### CODE REVIEW

File: pkg/sql/run_control_test.go [hunk 1]
```diff
@@ -326,7 +330,11 @@ GRANT admin TO has_admin2;
 					// end of the test.
 					errRE = "sql: database is closed"
```
Comment (otan) [inline pkg/sql/run_control_test.go:331]: i saw that typo ;)

**Review by otan — APPROVED** [review]

File: pkg/sql/run_control_test.go (hunk 1 above)
Comment (rafiss) [reply pkg/sql/run_control_test.go:331]: ur a wizard!

Comment (rafiss) [reply pkg/sql/run_control_test.go:331]: i did that to make sure the assertion was actually being called haha

**Comment by rafiss** [comment]
bors r=otan

**Pull request merged**